File: exchange/seed.py
from __future__ import annotations

import secrets

import bcrypt
from sqlalchemy.orm import Session

from exchange.config import SessionLocal, settings
from exchange.models import Account, Balance, Transaction


DEMO_BOTS = [
    {
        "bot_name": "SentimentBot",
        "developer_id": "dev-demo-1",
        "description": "Analyzes text sentiment.",
        "skills": ["sentiment-analysis", "text-classification"],
    },
    {
        "bot_name": "SummarizerBot",
        "developer_id": "dev-demo-2",
        "description": "Summarizes long-form content.",
        "skills": ["summarization", "text-extraction"],
    },
]


def seed(session: Session) -> None:
    print("Seeding demo accounts...")
    with session.begin():
        for bot in DEMO_BOTS:
            api_key = f"ate_{bot['bot_name'].lower()}_{secrets.token_hex(4)}"
            api_key_hash = bcrypt.hashpw(
                api_key.encode("utf-8"),
                bcrypt.gensalt(rounds=settings.api_key_salt_rounds),
            ).decode("utf-8")

            acct = Account(
                bot_name=bot["bot_name"],
                developer_id=bot["developer_id"],
                api_key_hash=api_key_hash,
                description=bot["description"],
                skills=bot["skills"],
            )
            session.add(acct)
            session.flush()

            session.add(Balance(account_id=acct.id, available=settings.starter_tokens))
            session.add(
                Transaction(
                    from_account=None,
                    to_account=acct.id,
                    amount=settings.starter_tokens,
                    tx_type="mint",
                    description="Starter token allocation (seed)",
                )
            )

            print(f"- {acct.bot_name}  id={acct.id}  api_key={api_key}")


def main() -> int:
    session = SessionLocal()
    try:
        seed(session)
    finally:
        session.close()
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: legacy/agent-token-exchange/migrations/seed.js
require('dotenv').config();
const { pool } = require('../config/database');
const bcrypt = require('bcrypt');
const { v4: uuidv4 } = require('uuid');

const STARTER_TOKENS = parseInt(process.env.STARTER_TOKEN_ALLOCATION || '100');

// Demo bots for testing the exchange
const demoBots = [
  {
    bot_name: 'SentimentBot',
    developer_id: 'dev-demo-1',
    description: 'Analyzes text sentiment. Returns confidence-scored positive/negative/neutral.',
    skills: ['sentiment-analysis', 'text-classification'],
  },
  {
    bot_name: 'SummarizerBot',
    developer_id: 'dev-demo-2',
    description: 'Produces concise summaries of long-form content.',
    skills: ['summarization', 'text-extraction'],
  },
  {
    bot_name: 'TranslatorBot',
    developer_id: 'dev-demo-3',
    description: 'Translates text between 40+ language pairs.',
    skills: ['translation', 'language-detection'],
  },
  {
    bot_name: 'CodeReviewBot',
    developer_id: 'dev-demo-4',
    description: 'Reviews code for bugs, security issues, and style.',
    skills: ['code-review', 'security-scan', 'linting'],
  },
  {
    bot_name: 'MessengerBot',
    developer_id: 'dev-rich',
    description: 'Orchestrator bot. Decomposes tasks and brokers skill requests.',
    skills: ['orchestration', 'task-decomposition'],
  },
];

async function seed() {
  const client = await pool.connect();
  console.log('Seeding demo data...');

  try {
    await client.query('BEGIN');

    for (const bot of demoBots) {
      const id = uuidv4();
      // Generate a readable API key for demo purposes
      const apiKey = `ate_${bot.bot_name.toLowerCase()}_${uuidv4().slice(0, 8)}`;
      const apiKeyHash = await bcrypt.hash(apiKey, 10);

      // Create account
      await client.query(
        `INSERT INTO accounts (id, bot_name, developer_id, api_key_hash, description, skills)
         VALUES ($1, $2, $3, $4, $5, $6)
         ON CONFLICT DO NOTHING`,
        [id, bot.bot_name, bot.developer_id, apiKeyHash, bot.description, JSON.stringify(bot.skills)]
      );

      // Create balance with starter tokens
      await client.query(
        `INSERT INTO balances (account_id, available)
         VALUES ($1, $2)
         ON CONFLICT DO NOTHING`,
        [id, STARTER_TOKENS]
      );

      // Record the mint transaction
      await client.query(
        `INSERT INTO transactions (from_account, to_account, amount, tx_type, description)
         VALUES (NULL, $1, $2, 'mint', 'Starter token allocation')`,
        [id, STARTER_TOKENS]
      );

      console.log(`  Created: ${bot.bot_name} (key: ${apiKey})`);
    }

    await client.query('COMMIT');
    console.log('Seed completed. Save the API keys above for testing!');
  } catch (err) {
    await client.query('ROLLBACK');
    console.error('Seed failed:', err.message);
    throw err;
  } finally {
    client.release();
    await pool.end();
  }
}

seed();
